### ISSUE TITLE
Escape sequences in chars not preserved

### ISSUE THREAD

**Issue by brekk** [madlib-lang/madlib#112]
Given an example like:

```madlib
import List from "List"
import String from "String"

escapeString :: String -> String
escapeString = (s) => {
  go :: List Char -> List Char
  go = (chars) => where(chars) {
    ['\\', '\n', ...cs] =>
      ['\\', '\n', ...go(cs)]

    [c, ...cs] =>
      [c, ...go(cs)]

    [] =>
      []
  }

  return pipe(
    String.toList,
    go,
    String.fromList,
  )(s)
}
```

Currently the automatic formatting doesn't seem to preserve the escape sequence properly, and produces autofixed code which is instead this:

```madlib
escapeString :: String -> String
escapeString = (s) => {
  go :: List Char -> List Char
  go = (chars) => where(chars) {
    ['\', ' ', ...cs] =>
      ['\', ' ', ...go(cs)]

    [c, ...cs] =>
      [c, ...go(cs)]

    [] =>
      []
  }

  return pipe(
    String.toList,
    go,
    String.fromList,
  )(s)
}
```
☝🏼 As `'\'` is an invalid string (incomplete), this then breaks things

**Comment by aboeglin** [comment]
Closed by #111

**Comment by brekk** [comment]
This was fixed a while ago but not closed; closing
